modules/renter/upload.go

```diff
@@ -102,6 +103,10 @@ func (r *Renter) Upload(up modules.FileUploadParams) error {
 	lockID := r.mu.Lock()
 	defer r.mu.Unlock(lockID)
 
+	if filepath.Ext(up.Filename) != filepath.Ext(up.Nickname) {
+		return errors.New("nickname and file name must have the same extension")
+	}
```

Comment:
this feels overly restrictive. How about instead we just autofill the Nickname field in the UI with the original filename? I think that's the behavior most people will want anyway.